plugins/adr/src/components/EntityAdrContent/EntityAdrContent.tsx

```diff
@@ -142,7 +147,11 @@ export const EntityAdrContent = (props: {
               </List>
             </Grid>
             <Grid item xs={9}>
-              <AdrReader adr={selectedAdr} decorators={contentDecorators} />
+              <AdrReader
+                adr={selectedAdr}
+                adrFileFetcher={targetAdrFileFetcher}
```

Comment:
Wouldn't this be problematic if you had URLs from multiple sources? I'm thinking it makes more sense to use the existing backend and rely on UrlReaders instead for all reading? That would support multiple locations from the same fetcher.

Reply:
I do indeed use UrlReaders for urls from multiple sites:  https://github.com/backstage/backstage/blob/0b1c9e4d651ed7267849dd08c565e82fbbc26912/plugins/adr-backend/src/service/router.ts#L34

I have expanded the adr backend plugin to provide two endpoints- one for reading a file at a provided url and another for reading a tree at a provided url. 

If you set the optional adrFileFetcher prop in EntityAdrContent to urlReaderAdrFileFetcher then it will switch to using UrlReaders. I have made it an option to use octokit or urlreaders in case people still want to use it the octokit way (for instance, if rate limiting was an issue). If that prop is not specified it will use octokit.